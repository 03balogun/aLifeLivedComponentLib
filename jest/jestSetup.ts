/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
