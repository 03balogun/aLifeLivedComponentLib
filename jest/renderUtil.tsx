import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
