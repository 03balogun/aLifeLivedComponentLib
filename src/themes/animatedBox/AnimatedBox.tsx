import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  color,
  ColorProps,
  createRestyleComponent,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from '@shopify/restyle';
import React from 'react';
import { Animated, ViewProps } from 'react-native';
import { Theme } from '../theme';

export type StyledOpacityProps = Animated.AnimatedProps<ViewProps> &
  SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> &
  ColorProps<Theme> & { children?: React.ReactNode };

export const AnimatedBox = createRestyleComponent<StyledOpacityProps, Theme>(
  [spacing, layout, border, backgroundColor, color],
  Animated.View
);
