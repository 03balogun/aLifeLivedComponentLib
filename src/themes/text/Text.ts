import { createText } from '@shopify/restyle';
import { Theme } from '../theme';

export const textVariants = {
  h1: {
    fontSize: 32,
    lineHeight: 38,
    color: 'contrastHighest',
  },
  h2: {
    fontSize: 24,
    lineHeight: 30,
    color: 'contrastHighest',
  },
  h3: {
    fontSize: 16,
    lineHeight: 24,
    color: 'contrastHighest',
  },
  body: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
  caption: {
    fontSize: 14,
    lineHeight: 18,
    color: 'contrastLow',
  },
  default: {
    fontSize: 16,
    lineHeight: 20,
    color: 'contrastHighest',
  },
};

export const Text = createText<Theme>();
