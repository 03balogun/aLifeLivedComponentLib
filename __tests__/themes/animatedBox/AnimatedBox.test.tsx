import React from 'react';
import { AnimatedBox } from '../../../src/themes/animatedBox/AnimatedBox';
import { render } from '../../../jest/renderUtil';
import { Text } from '../../../src/themes/text/Text';

describe('themes >> animatedBox', () => {
  describe('Should render animated box with text inside', () => {
    test('Render snapshot correctly', () => {
      const { toJSON } = render(
        <AnimatedBox>
          <Text variant="h1">H1 variant</Text>
        </AnimatedBox>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
