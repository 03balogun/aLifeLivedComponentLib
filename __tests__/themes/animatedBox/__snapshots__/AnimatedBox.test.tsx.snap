// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> animatedBox Should render animated box with text inside Render snapshot correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "display": "flex",
          "fontSize": 32,
          "lineHeight": 38,
        },
      ]
    }
  >
    H1 variant
  </Text>
</View>
`;
